variables.tf

```
@@ -43,11 +43,11 @@ variable "access_group_roles" {
   default     = null
   validation {
     error_message = "Invalid role set for the access group, all roles must be one of: Reader, Writer, Manager, SecretsReader, Viewer, Operator, Editor, Administrator, Service Configuration Reader, Key Manager"
-    condition     = (var.access_group_roles != null && length(setintersection(var.access_group_roles, ["Reader", "Writer", "Manager", "SecretsReader", "Viewer", "Operator", "Editor", "Administrator", "Service Configuration Reader", "Key Manager"])) != 0)
+    condition     = (var.access_group_roles == null ? true : length(setintersection(var.access_group_roles, ["Reader", "Writer", "Manager", "SecretsReader", "Viewer", "Operator", "Editor", "Administrator", "Service Configuration Reader", "Key Manager"])) != 0)
   }
   validation {
     error_message = "When creating an access group, a list of roles must be passed"
-    condition     = (var.create_access_group && var.access_group_roles != null && length(var.access_group_roles) != 0)
+    condition     = (var.access_group_roles == null ? !var.create_access_group : (var.create_access_group && length(var.access_group_roles) != 0))
```

I am not convinced the logic is right here.

If create_access_groups is true && number of roles > 0 then it is right.

```
condition = var.create_access_group && length(var.access_group_rules) > 0
```

That condition should result in true and avoid the error message.

Then either:
Add another validation that says roles MUST be null if create is false.
Or just silently ignore the roles is the request is not to create them (I am not sure this is a good solution)

```
validation {
error_message = "Roles should only be specified when create_access_group is set to true"
condition = !var.create_access_group && length(var.access_group_rules) == 0
```

---

The first suggestion doesn't work if access_group_roles is null.